### PR TITLE
Bundle quick wins: #35 #38 #43 #50 #51

### PR DESCRIPTION
**Pull request by jasonherald**
## Summary

Bundles five "trivial / small" polish-pass items from epic #29 into one PR:

- **#43** — Extracted `NotificationState::next_notification_id()` with a docstring covering the wrap-around protection and the freedesktop `id != 0` invariant.
- **#35** — Lifted the three nerd-font icon codepoints in `src/waybar.rs` to named constants (`ICON_BELL_OFF`, `ICON_BELL_BADGE`, `ICON_BELL_OUTLINE`); added a module docstring documenting the nerd-font dependency.
- **#50** — Dropped the unused `nix = "0.31"` direct dep (we have zero `use nix` statements; nwg-common pulls in its own slice). `clap = ["derive"]` was already minimal — no change needed there. Also tidied the stale `nwg-common` comment that still said "0.3.x line".
- **#38** — Tightened every `pub` in the bin crate to `pub(crate)` — 125 items across 17 files via line-anchored sed. No `pub use` / `pub trait` / `pub(super)` edge cases. No `tests/` directory means nothing reaches across the bin-crate boundary.
- **#51** — No code change. Confirmed via CHANGELOG note that `nwg-common = "0.4"` (caret = `>=0.4.0, <0.5.0`) already restricts to the 0.4.x train, which is what the issue asked for.

One commit per issue, smallest blast radius first. Plan committed first under `docs/superpowers/plans/`.

## Test plan

- [x] `make lint` clean locally (fmt + clippy + test + deny + audit) on every commit.
- [x] Existing 68 unit tests still green; no new tests were warranted (all changes are docs / mechanical / dep-removal).
- [x] Manual smoke test against the live compositor (installed via `make upgrade PREFIX=\$HOME/.local BINDIR=\$HOME/.cargo/bin`):
  - [x] `notify-send` still produces a popup.
  - [x] Panel still renders, nerd-font glyphs render correctly.
  - [x] DND toggle still works (bell-off glyph renders).
  - [x] `nwg-notifications --update --max-history N` still works (D-Bus surface intact after pub→pub(crate)).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Chores**
  * Updated dependencies to target `nwg-common` 0.4.x and removed unused `nix` dependency.

* **Refactor**
  * Restricted internal codebase visibility to reduce public API surface.
  * Refactored notification ID allocation with improved wrapping behavior.
  * Extracted icon glyphs into named constants for consistency.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 > The full pre-split history is preserved in the monorepo's git log; this
 > file only documents changes from v0.3.0 onward.
 
+## [Unreleased]
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply CHANGELOG.md]: @coderabbitai you're right — the right fix turned out to be one level deeper than the header format alone. Per your other finding (3178847856), the entire bundle is internal cleanup with no user-visible impact, so the [Unreleased] section shouldn't exist at all. Removed in 449e192. If/when something user-visible is queued before the next release, a fresh `## [x.y.z] — Unreleased` section will go in then.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Code-quality polish pass: tightened every `pub` item in the bin
+  crate to `pub(crate)` (#38), lifted the three nerd-font icon
+  codepoints in `waybar.rs` to named constants with a module
+  docstring documenting the font dependency (#35), extracted a
+  documented `next_notification_id()` helper that records the
+  freedesktop `id != 0` invariant (#43), and dropped the unused
+  `nix` direct dep — `clap` was already at the minimum
+  `["derive"]` feature set (#50). Confirmed that
+  `nwg-common = "0.4"` (caret = `>=0.4.0, <0.5.0`) already does
+  what a manual 0.4.x pin would; no code change needed (#51).
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply CHANGELOG.md]: @coderabbitai agreed completely. None of the five issues in this bundle (#35 #38 #43 #50 #51) change behavior, output, the public CLI/D-Bus surface, or anything a user would notice from running the daemon. The dep-removal in #50 is theoretically visible to a downstream packager scanning Cargo.toml, but that's already self-evident from the diff and doesn't warrant a CHANGELOG entry per Keep-a-Changelog's "what changed for the user" framing.

Removed the entire `## [Unreleased]` section in 449e192. The audit trail lives in the PR description, the closed-issue links (each commit ends with a `Closes #NN` trailer), and the per-commit messages — that's the right place for internal-cleanup history. A fresh `## [x.y.z] — Unreleased` section will be added if/when there's something user-visible queued before the next release.

File: docs/superpowers/plans/2026-05-03-bundle-quick-wins-35-38-43-50-51.md [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+---
+
+## File Structure
+
+| Task | Files modified | Test approach |
+|------|----------------|---------------|
+| #43 `id != 0` invariant | `src/state.rs` (extract `next_notification_id`) | Existing `id_wrapping_at_max` test still passes (now exercises the helper) |
+| #35 nerd-font constants | `src/waybar.rs` (3 `const`s + module docstring) | Build only — visible glyphs unchanged |
+| #50 dep audit | `Cargo.toml` (drop unused `nix`; tidy stale `nwg-common` comment), `Cargo.lock` (regen) | `cargo build --release` succeeds with reduced dep set |
+| #38 `pub` → `pub(crate)` | All `src/**/*.rs` files with `pub ` items (line-anchored sed) | `cargo build --release` + `cargo test` still green |
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply docs/superpowers/plans/2026-05-03-bundle-quick-wins-35-38-43-50-51.md]: @coderabbitai both addressed in 96faab9.

- **MD038**: the `\`pub \`` (with trailing space inside the backticks) was meant to convey "the literal pub-keyword-followed-by-space pattern that the sed sweep matches", but the trailing space is exactly what the rule prohibits. Reworded the table cell + the four other instances I found elsewhere in the doc to `pub`-prefixed / module-level / field-level — same meaning, no trailing space.
- **MD031**: the `\`\`\`rust` fence on line 73 opened immediately after the "Expected output should show..." narrative line. Inserted a blank line before it.

`markdownlint-cli2` now reports zero MD038 and zero MD031 findings on the file.

File: docs/superpowers/plans/2026-05-03-bundle-quick-wins-35-38-43-50-51.md [hunk 4]
```diff
@@ -0,0 +1,614 @@
...
+---
+
+## Task 3: #50 — Drop the unused `nix` dep + tidy the stale `nwg-common` comment
+
+**Files:**
+- Modify: `Cargo.toml` — remove the `nix` line entirely; update the now-stale comment above `nwg-common`.
+- Modify: `Cargo.lock` — regenerated by `cargo build`.
+
+**Investigation (already done; baked into this task):**
+- `grep -rn "use nix\|nix::" src/` returns one line — a comment in `src/listeners.rs:31` explaining that nwg-common's signal helpers drop to raw libc because nix's `Signal` enum doesn't cover RT signals. We have no `use nix` statement and no `nix::` call sites of our own. The `nix = "0.31"` direct dep with three features pulled in is dead weight — nwg-common pulls its own slice of `nix` transitively.
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply docs/superpowers/plans/2026-05-03-bundle-quick-wins-35-38-43-50-51.md]: @coderabbitai exactly the kind of rot you flagged in the prior learning. Replaced `src/listeners.rs:31` with "an inline comment near the top of `start_signal_listener` in `src/listeners.rs`" — anchored to the function name, not the line number. Fixed in 96faab9.

**Comment by jasonherald** [comment]
@coderabbitai re. the outside-diff finding on `src/state.rs:264-286` (strengthen `id_wrapping_at_max`):

You're right — the original test verified the wrapping arithmetic in isolation plus a happy-path two-add sequence starting from `next_id == 1`. The actual wrap through the `add()` → `next_notification_id()` path was untested, so the freedesktop "id != 0" invariant could quietly regress and the old test would still pass.

Applied your suggested patch in cec4e43 with one extra add() to also confirm the post-wrap sequence continues at 2:

```rust
state.next_id = u32::MAX;
let id_max = state.add(test_notif("app", "max"));
let id_wrapped = state.add(test_notif("app", "wrapped"));
let id_after = state.add(test_notif("app", "after"));
assert_eq!(id_max, u32::MAX);
assert_eq!(id_wrapped, 1);
assert_eq!(id_after, 2);
assert_ne!(id_max, 0);
assert_ne!(id_wrapped, 0);
assert_ne!(id_after, 0);
```

Direct field assignment works because `next_id` is `pub(crate)`-less (private) and the test lives in the same module. Test still passes; full state suite (16 tests) still green.
